control-plane/agents/src/bin/core/main.rs

```diff
@@ -103,6 +103,10 @@ pub(crate) struct CliArgs {
     /// Events message-bus endpoint url.
     #[clap(long, short)]
     events_url: Option<url::Url>,
+
+    /// Flag to track whether HA is enabled or not.
+    #[clap(long, default_value = "true", env = "HA_ENABLED")]
+    pub(crate) ha_enabled: bool,
```

Comment:
It's bit confusing to have enabled bool by default IMHO, so suggest something like this:

```suggestion
    /// Disable the HA/Failover feature. 
    /// This is useful when the frontend nodes do not support the NVMe ANA feature.
    #[clap(long, env = "HA_DISABLED")]
    pub(crate) disable_ha: bool,
```

Reply:
The problem is then we would need to negate and pass the value from helm template. I kept it so that we can pass the value as it is.

Reply:
What's the problem with negating the value in helm?

Reply:
No there is no problem, it's just that helm follows the enabled keyword flow, so I did that way initially.

Reply:
Anyway, not a deal breaker to me, just a nit, we can keep it but please enhance doc comment.

Reply:
No, if we look from the cli args perspective I think the disable makes more sense here. I will make the change.